lib/ts/recipe/passwordless/recipeImplementation.ts

```diff
@@ -0,0 +1,227 @@
+                if (validationRes !== undefined) {
+                    return {
+                        status: "GENERAL_ERROR",
+                        message: validationRes,
+                    };
+                }
+                bodyObj = {
+                    email: input.email,
+                };
+            }
+            if ("phoneNumber" in input) {
+                const validationRes = await input.config.validatePhoneNumber(input.phoneNumber);
+                if (validationRes !== undefined) {
+                    return {
+                        status: "GENERAL_ERROR",
+                        message: validationRes,
+                    };
+                }
+                bodyObj = {
+                    phoneNumber: input.phoneNumber,
+                };
+            }
+
+            return await querier.post("/signinup/code", { body: JSON.stringify(bodyObj) }, (context) => {
+                return input.config.preAPIHook({
+                    ...context,
+                    action: "PASSWORDLESS_CREATE_CODE",
+                });
+            });
+        },
+        resendCode: async function (
+            input: { deviceId: string; preAuthSessionId: string } & {
+                config: NormalisedConfig;
+            }
+        ): Promise<ResendCodeApiResponse> {
+            const bodyObj = {
+                deviceId: input.deviceId,
+                preAuthSessionId: input.preAuthSessionId,
+            };
+
+            return await querier.post("/signinup/code/resend", { body: JSON.stringify(bodyObj) }, (context) => {
+                return input.config.preAPIHook({
+                    ...context,
+                    action: "PASSWORDLESS_RESEND_CODE",
+                });
+            });
+        },
+        consumeCode: async function (
+            input: (
+                | {
+                      userInputCode: string;
+                      deviceId: string;
+                      preAuthSessionId: string;
+                  }
+                | {
+                      preAuthSessionId: string;
+                      linkCode: string;
+                  }
+            ) & {
+                config: NormalisedConfig;
+            }
+        ): Promise<ConsumeCodeApiResponse> {
+            let bodyObj;
+            if ("userInputCode" in input) {
+                // TODO: validate input code?
+                bodyObj = {
+                    userInputCode: input.userInputCode,
+                    deviceId: input.deviceId,
+                    preAuthSessionId: input.preAuthSessionId,
+                };
+            }
+            if ("linkCode" in input) {
+                bodyObj = {
+                    linkCode: input.linkCode,
+                    preAuthSessionId: input.preAuthSessionId,
+                };
+            }
+
+            return await querier.post("/signinup/code/consume", { body: JSON.stringify(bodyObj) }, (context) => {
```

Comment:
You need to check if this was successful. If it was, you need to fire the relevant event using `onHandleEvent`